mwdb/web/src/commons/ui/ObjectTab.tsx

```diff
@@ -17,24 +19,61 @@ type Props = {
     icon?: IconProp;
     tab: string;
     label?: string | JSX.Element;
+    dropdownActions?: boolean;
 };
 
 export function ObjectTab(props: Props) {
     const context: TabContextValues = useTabContext();
 
     useEffect(() => {
         if (context.tab !== props.tab) return;
+
         context.setComponent(props.component);
-        context.setActions(
-            props.actions?.map((action, index) => {
-                return {
-                    ...action,
-                    key: index,
-                };
-            }) || []
-        );
+
+        const previewSwitchAction =
+            props.actions?.find(
+                (action) => action.type === PreviewSwitchAction
+            ) || null;
+
+        const otherActions =
+            props.actions?.filter(
+                (action) => action.type !== PreviewSwitchAction
+            ) || [];
+
+        const actionsArray: JSX.Element[] = [];
+        if (previewSwitchAction) {
+            actionsArray.push(
+                <React.Fragment key="preview-switch">
+                    {previewSwitchAction}
+                </React.Fragment>
+            );
+        }
+        if (otherActions.length > 0 && props.dropdownActions !== false) {
+            actionsArray.push(
+                <NavDropdown
+                    key="nav-dropdown"
+                    title="Action"
```

Comment:
I would add plural there
```suggestion
                    title="Actions"
```